lib/migration.js

```diff
@@ -9,35 +9,43 @@ var async = require('async');
 module.exports = mixinMigration;
 
 function mixinMigration(Oracle) {
+  Oracle.prototype.showFields = function(model, cb) {
+    var sql = 'SELECT column_name AS "column", data_type AS "type",' +
+      ' nullable AS "nullable"' + // , data_default AS "Default"'
+      ' FROM "SYS"."USER_TAB_COLUMNS" WHERE table_name=\'' +
+      this.table(model) + '\'';
+    this.execute(sql, function(err, fields) {
+      if (err)
+        return cb(err);
+      else
+        cb(err, fields);
+    });
+  };
+
   /*!
    * Discover the properties from a table
    * @param {String} model The model name
    * @param {Function} cb The callback function
    */
   Oracle.prototype.getTableStatus = function(model, cb) {
-    function decoratedCallback(err, data) {
-      if (err) {
-        console.error(err);
-      }
-      if (!err) {
-        data.forEach(function(field) {
-          field.type = mapOracleDatatypes(field.type);
-        });
-      }
-      cb(err, data);
-    }
+    var fields;
+    var self = this;
 
-    this.execute('SELECT column_name AS "column", data_type AS "type",' +
-      ' nullable AS "nullable"' + // , data_default AS "Default"'
-      ' FROM "SYS"."USER_TAB_COLUMNS" WHERE table_name=\'' +
-      this.table(model) + '\'', decoratedCallback);
+    this.showFields(model, function(err, data) {
+      if (err) return cb(err);
+      fields = data;
+
+      if (fields)
+        return cb(null, fields);
+    });
   };
+
   /**
-   * Perform autoupdate for the given models
-   * @param {String[]} [models] A model name or an array of model names. If not
-   * present, apply to all models
-   * @param {Function} [cb] The callback function
-   */
+  * Perform autoupdate for the given models
+  * @param {String[]} [models] A model name or an array of model names. If not
+  * present, apply to all models
+  * @param {Function} [cb] The callback function
+  */
   Oracle.prototype.autoupdate = function(models, cb) {
```

Comment:
This comment might be too earlier, just a reminder, we should remove the function after refactor done :)